File: WowsKarma.Common/Models/DTOs/PostModActionDTO.cs
namespace WowsKarma.Common.Models.DTOs;

public record PostModActionDTO
{
	public Guid Id { get; init; }

	[Required]
	public Guid PostId { get; init; }

	public PlayerPostDTO? UpdatedPost { get; init; }

	public ModActionType ActionType { get; init; }

	public uint ModId { get; init; }
	public string ModUsername { get; init; } = string.Empty;

	[Required]
	public string Reason { get; set; } = string.Empty;
}

File: wowskarma.app/src/app/services/api/models/post-mod-action-dto.ts
/* tslint:disable */
/* eslint-disable */
import { ModActionType } from './mod-action-type';
import { PlayerPostDto } from './player-post-dto';
export interface PostModActionDto {
  actionType?: ModActionType;
  id?: string;
  modId?: number;
  modUsername?: null | string;
  postId: string;
  reason: string;
  updatedPost?: PlayerPostDto;
}
